File: 2019/01/1.jl
f = open("2019/01/input.txt", "r")
inlines = [parse(Int, x) for x in readlines(f)]
close(f)

outlines = [[(x÷3)-2] for x in inlines]

while true
    going = false
    for l in outlines
        fuel = (last(l)÷3)-2
        if fuel > 0
            #println(fuel)
            push!(l, fuel)
            going = true
        end
    end
    if !going
        break
    end
end

println(sum([sum(l) for l in outlines]))

File: 2021/10/1.py
#!/usr/bin/python3

points = {
    ')': 3, 
    ']': 57, 
    '}': 1197, 
    '>': 25137
}

match = {
    '(': ')',
    '[': ']',
    '{': '}',
    '<': '>'
}

input = []
with open('10/input.txt', 'r') as f:
    for line in f.readlines():
        input.append(line.strip())

losers = []
for line in input:
    stack = []
    for i in range(0, len(line)):
        c = line[i]
        if c in match.keys():
            stack.append(c)
        elif c == match[stack[-1]]:
            stack.pop()
        else:
            if i != len(line) - 1:
                losers.append(c)
                break

sum = 0
for loser in losers:
    sum += points[loser]
print(sum)